### PR TITLE
Evaluating Machine Learning Model Performance

### PR DESCRIPTION
**Pull request by collins-a**
Article as index.md
Hero image as hero.jpg

Additional images:
fscore.jpg, mae.jpg, mse.jpg, confusion.png, perecisionrecall.png

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by collins-a** [comment]
Fixes #633

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by hectorkambow** [comment]
[Evaluating Machine Learning Model Performance #675.pdf](https://github.com/section-io/engineering-education/files/5584977/Evaluating.Machine.Learning.Model.Performance.675.pdf)
@collins-a 👋  Good morning - just an FYI - upon running your article through our 3rd party plagiarism checker it raised a couple red flags ( a few more than we like to see) nothing too major.
Please go over the PDF and check if any extra sentences need extra links or sources. 
👍  
Thank you

**Comment by collins-a** [comment]
Hello @hectorkambow, apologies for that and thank you for the heads up. Let me sort it out asap 😁

**Comment by collins-a** [comment]
@hectorkambow hey! I've added a number of links to the flagged sections that were of relevance to the article. I'll be keener next time. Let me know if I've done the correct thing or if there are more changes to be made

File: articles/evaluating-ml-model-performance/index.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+The mean squared error is quite similar to the mean absolute error. However, as described by this [article](https://medium.com/analytics-vidhya/understanding-performance-metrics-for-machine-learning-algorithms-996dd7efde1e), mean squared error uses the average of the square of the difference between original and predicted values. Since this involves the squaring of the errors, larger errors are very notable.
+Mean squared error is given by:
+
+![mse](/engineering-education/evaluating-ml-model-performance/mse.jpg)
+
+[Source](https://towardsdatascience.com/metrics-to-evaluate-your-machine-learning-algorithm-f10ba6e38234)
+
+##### Root Mean Squared Error
+
```
Comment (lalith1403) [inline articles/evaluating-ml-model-performance/index.md:138]: Why is RMSE used? Mention that as well. (Smoother gradients and L2 norm).

File: articles/evaluating-ml-model-performance/index.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+##### Classification Accuracy
+
+The most common evaluation metric for classification problems is accuracy. It is taken as the number of correct predictions against the total number of predictions made (or input samples). However, as much as accuracy is used to evaluate a model, it is not a clear indicator of model performance.
+Classification accuracy works best if the samples belonging to each class are equal in number. Consider a scenario with 97% samples of class X and 3% class Y in a training set. A model can very easily achieve 97% training accuracy by predicting each training sample in class X.
+Testing the same model on a test set with 55% samples of X and 45% samples of Y, the test accuracy is reduced to 55%. This is why classification accuracy is not a clear indicator of performance. It provides a false sense of attaining high levels of accuracy.
+
+##### Confusion Matrix
+
+The confusion matrix forms the basis for the other types of classification metrics. It is a matrix that fully describes the performance of the model. A confusion matrix gives an in-depth breakdown of the correct and incorrect classifications of each class.
```
Comment (lalith1403) [inline articles/evaluating-ml-model-performance/index.md:66]: In a confusion matrix, what values are acceptable? Is it alright to have false positives? Is it alright to have true negatives? Which ones should be minimized? What insights does one get from CM? Answer these questions. Moreover, define precision and recall intuitively. In terms of the memory property of the model.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
A few changes and this should be good to go. A Grammarly check after edits is advisable to ensure minimal rounds of revision.

**Comment by collins-a** [comment]
@lalith1403 hey, I've made some additions but I struggled when it came to what is meant by the memory property of the model as well as on the part of normalization. I might need some time for some personal research on normalization and regularization. Thanks for the challenge (and a potential new topic to write on). However, for the sake of the article, I've altered and added what I can at this moment. I've included a link to an article on RMSE to give further information in addition to the paragraph I added. 

Done a Grammarly check as well. The few alerts that remain are basically false positives.

Let me know if these changes are adequate. Thanks again!

**Review by lalith1403 — APPROVED** [review]

**Pull request merged**